src/main/java/io/conduit/SourceStream.java

```diff
@@ -16,127 +16,21 @@
-                        Function<SourceRecord, Record.Builder> transformer) {
-        this.task = task;
-        this.position = position;
-        this.responseObserver = responseObserver;
-        this.transformer = transformer;
-    }
-
-    @Override
-    public void run() {
-        while (shouldRun) {
-            try {
-                if (buffer.isEmpty()) {
-                    fillBuffer();
-                }
-                SourceRecord rec = buffer.poll();
-                // We may get so-called tombstone records, i.e. records with a null payload.
-                // This can happen when records are deleted, for example.
-                // This is used in Kafka Connect internally, more precisely for log compaction in Kafka.
-                // For more info: https://kafka.apache.org/documentation/#compaction
-                if (rec.value() != null) {
-                    responseObserver.onNext(responseWith(rec));
-                }
-            } catch (Exception e) {
-                Logger.get().error("Couldn't write record.", e);
-                responseObserver.onError(
-                        Status.INTERNAL
-                                .withDescription("couldn't read record: " + e.getMessage())
-                                .withCause(e)
-                                .asException()
-                );
-            }
-        }
-        Logger.get().info("SourceStream loop stopped.");
-    }
-
-    @Override
-    public void onNext(Source.Run.Request value) {
-        // todo Acknowledging record not implemented yet...
-        // See: https://github.com/ConduitIO/conduit-kafka-connect-wrapper/issues/59
-    }
-
-    @SneakyThrows
-    private void fillBuffer() {
-        List<SourceRecord> polled = task.poll();
-        while (Utils.isEmpty(polled)) {
-            polled = task.poll();
-        }
-        buffer.addAll(polled);
-    }
-
-    @SneakyThrows
-    private Source.Run.Response responseWith(SourceRecord rec) {
-        position.add(rec.sourcePartition(), rec.sourceOffset());
-
-        Record.Builder conduitRec = transformer.apply(rec)
-                .setPosition(position.asByteString());
-
-        return Source.Run.Response.newBuilder()
-                .setRecord(conduitRec)
-                .build();
-    }
-
-    @Override
-    public void onError(Throwable t) {
-        Logger.get().error("Experienced an error.", t);
-        stop();
-        responseObserver.onError(
-                Status.INTERNAL.withDescription("Error: " + t.getMessage()).withCause(t).asException()
-        );
-    }
-
-    @Override
-    public void onCompleted() {
-        Logger.get().info("Completed.");
-        stop();
-        responseObserver.onCompleted();
-    }
-
-    private void stop() {
-        Logger.get().info("Stopping...");
-        shouldRun = false;
-    }
+public interface SourceStream extends StreamObserver<Source.Run.Request>, Runnable {
```

Comment:
SourceStream was previously a class, not it's an interface. The default implementation is `DefaultSourceStream`.